### PR TITLE
Improvement: Upgrade electron from 8.x.y to 11.5.0 and improve to render multi toolpath in parallel using workerpool

### PR DESCRIPTION
**Pull request by jane-rose**

### CODE REVIEW

File: src/app/flux/editor/index.js [hunk 1]
```diff
@@ -214,8 +213,11 @@ function recordScaleActionsToHistory(scaleActionsFn, elements, SVGActions, headT
         });
     }
 }
+const pool = workerpool.pool('ToolpathRenderer.worker.js', {
+    minWorkers: 'max',
+});
+console.log('onmessage', onmessage);
```
Comment (xiekun1992) [inline src/app/flux/editor/index.js:219]: remove log

File: src/main.js [hunk 2]
```diff
@@ -37,7 +37,11 @@ function getBrowserWindowOptions() {
         useContentSize: true,
         title: `${pkg.name} ${pkg.version}`,
         webPreferences: {
-            nodeIntegration: true
+            nodeIntegration: true,
+            contextIsolation: false,
```
Comment (xiekun1992) [inline src/main.js:41]: do we need so many settings?

**Review by xiekun1992 — APPROVED** [review]

**Review by bbaa77770 — APPROVED** [review]

File: src/app/flux/editor/actions-process.js [hunk 3]
```diff
@@ -29,8 +28,14 @@ export const processActions = {
         }));
 
         // start generate toolpath
+        const toolPathPromiseArray = [];
```
Comment (MaShizhen) [inline src/app/flux/editor/actions-process.js:31]: can use Array.map

File: src/app/flux/editor/index.js [hunk 4]
```diff
@@ -2,9 +2,8 @@ import * as THREE from 'three';
 import path from 'path';
 import { v4 as uuid } from 'uuid';
 import _, { includes } from 'lodash';
-
-import ToolpathRendererWorker from '../../workers/ToolpathRenderer.worker';
-
+import workerpool from 'workerpool';
+import onmessage from '../../workers/ToolpathRenderer.worker';
```
Comment (MaShizhen) [inline src/app/flux/editor/index.js:6]: For similar scenarios, you can use `import 'module'`

**Review by MaShizhen — APPROVED** [review]

**Pull request merged**